### PR TITLE
Adding new files based on an old tag

### PR DESCRIPTION
**Pull request by lucasncabral**
When adding files based on a different tag from the most recent one, ml-git takes all the data from the other tags into that new tag.

**Scenario:** 
Create a new entity
Add 1 file (let's call it A)
Commit and push (this will create tag 1)
Add 1 more file (let's call it B)
Commit and push (this will create tag 2)
Checkout the tag 1
Add 1 file (let's call C)
Commit with --version=3 and push (this will create tag 3)

**Expected behavior:**
  Checkout tag 3 - files A and C are downloaded.

**Observed behavior:**
  Checkout tag 3 - files A, B and C are downloaded.

### CODE REVIEW

**Review by gilMars — APPROVED** [review]

**Review by Mun1z — APPROVED** [review]

File: ml_git/_metadata.py [hunk 1]
```diff
@@ -313,8 +313,13 @@ def get_metadata_manifest(self, path):
             return Manifest(path)
         return None
 
-    def remove_deleted_files_meta_manifest(self, manifest, deleted_files):
+    def remove_deleted_files_meta_manifest(self, manifest, ws_path):
+        deleted_files = []
```
Comment (diego-pedroso) [inline ml_git/_metadata.py:317]: Does this list differ from the one generated by `Objects.commit_index`?

https://github.com/HPInc/ml-git/blob/b1d46d768fd7213d82fd258ad5f25bf938e9195d/ml_git/repository.py#L348

File: ml_git/repository.py [hunk 2]
```diff
@@ -349,9 +349,9 @@ def commit(self, spec, specs, version=None, run_fsck=False, msg=None):
 
         bare_mode = os.path.exists(os.path.join(index_path, 'metadata', spec, 'bare'))
 
-        if not bare_mode and len(deleted_files) > 0:
-            self.remove_deleted_files(idx, index_path, m, manifest_path, spec, deleted_files)
-        elif bare_mode:
+        if not bare_mode:
```
Comment (diego-pedroso) [inline ml_git/repository.py:352]: I think `commit()` is way too complex. We could break it down into smaller ones in the future.

File: ml_git/repository.py [hunk 3]
```diff
@@ -371,12 +371,12 @@ def commit(self, spec, specs, version=None, run_fsck=False, msg=None):
         return tag
 
     @Halo(text='Checking removed files', spinner='dots')
-    def remove_deleted_files(self, idx, index_path, m, manifest_path, spec, deleted_files):
+    def remove_deleted_files(self, idx, index_path, m, manifest_path, spec, deleted_files, ws_path):
```
Comment (diego-pedroso) [inline ml_git/repository.py:374]: Is this used anywhere else? If not, please add a leading underscore. Also, please declare it before it is used.

File: ml_git/_metadata.py (hunk 1 above)
Comment (lucasncabral) [reply ml_git/_metadata.py:317]: Yes. In fact, considering these lists to be the same was what was causing the inappropriate behavior reported in the bug.

File: ml_git/repository.py (hunk 3 above)
Comment (lucasncabral) [reply ml_git/repository.py]: Done in 9523ed1.

File: ml_git/_metadata.py (hunk 1 above)
Comment (diego-pedroso) [reply ml_git/_metadata.py:317]: How is that so? Is there any way we can make it clearer in the code?

Comment (lucasncabral) [reply ml_git/_metadata.py]: The list of deleted files returned by `Objects.commit_index` is generated based on the file located in .ml-git\dataset|labels|model)\index. This list captures files that have been deleted based on the checkout tag, while in the `_metadata.remove_deleted_files_meta_manifest` method, files related to other tags that have been added after the base tag also are deleted.

I think the misunderstanding is being generated by treating files added in tags after the base tag as excluded files. To solve this, the code was separated to better define these steps. This change was applied in 4cb92e3.

**Review by diego-pedroso — APPROVED** [review]

**Pull request merged**